dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/RepositoryUtility.java

```diff
@@ -65,17 +69,44 @@ public static RepositorySystem newRepositorySystem() {
   }
 
   /**
-   * Open a new Maven repository session that looks for the local repository in the
+   * Opens a new Maven repository session that looks for the local repository in the
    * customary ~/.m2 directory. If not found, it creates an initially empty repository in
    * a temporary location.
    */
-  public static RepositorySystemSession newSession(RepositorySystem system ) {
+  public static RepositorySystemSession newSession(RepositorySystem system) {
     DefaultRepositorySystemSession session = MavenRepositorySystemUtils.newSession();
   
     LocalRepository localRepository = new LocalRepository(findLocalRepository().getAbsolutePath());
     session.setLocalRepositoryManager(system.newLocalRepositoryManager(session, localRepository));
     session.setReadOnly();
-    
+
+    return session;
+  }
+
+  /**
+   * Opens a new Maven repository session in the same way as {@link
+   * RepositoryUtility#newSession(RepositorySystem)}, with its dependency selector to include
+   * dependencies with 'provided' scope.
+   */
+  public static RepositorySystemSession newSessionWithProvidedScope(RepositorySystem system) {
+    DefaultRepositorySystemSession session = MavenRepositorySystemUtils.newSession();
```

Comment:
Let's create a private method that returns the same thing as `newSession()` but without setting readonly.

Then `newSession()` can just call `setReadOnly()`, and this method can just add the other configuration first, and we don't have to copy any code.